packages/format-library/src/text-color/inline.native.js

```diff
@@ -110,7 +104,9 @@ function setColors( value, name, colorSettings, colors ) {
 		const newFormat = applyFormat( value, format );
 		const { activeFormats } = newFormat;
 		newFormat.formats[ value.start ] = [
-			...reject( activeFormats, { type: format.type } ),
+			...( activeFormats?.filter(
+				( { type } ) => type !== format.type
+			) || [] ),
```

Comment:
I checked this change in the native version and confirm it's working as expected 🎊 .

Reply:
Thanks for the confirmation @fluiddot 🙌